src/event.rs

```diff
@@ -353,9 +353,27 @@ pub enum StartCause {
     Init,
 }
 
+/// Represents different attributes that can be associated with a [`Window`].
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum WindowAttribute {
+    /// The window is in normal mode
+    Normal,
+    /// The window is in fullscreen mode
+    Fullscreen,
+    /// The window is minimized
+    Minimized,
+}
+
 /// Describes an event from a [`Window`].
 #[derive(Debug, PartialEq)]
 pub enum WindowEvent<'a> {
+    /// An event which represents a change in the attribute of a [`Window`].
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **iOS / Android / Web / Orbital, X11, Wayland:** Unsupported.
```

Comment:
For X11 I would assume we can use `UnmapNotify `and `MapNotify ` for minimized state, however I can't help work on it since I don't have a Linux machine at hand.